src/test/shared/utilities/vsCodeUtils.test.ts

```diff
@@ -0,0 +1,164 @@
+                        // Test channel write
+                        assert(
+                            outputChannel.value.indexOf(expectedPrettyMsg) >= 0,
+                            `channel missing msg: ${expectedPrettyMsg} in ${outputChannel.value}`
+                        )
+
+                        // Test processTemplate
+                        const {
+                            prettyMessage: actualPrettyMsg,
+                            errors: actualErrorTokens
+                        } = processTemplate({
+                            nlsKey,
+                            nlsTemplate,
+                            templateTokens
+                        })
+                        assert(
+                            expectedPrettyMsg === actualPrettyMsg,
+                            `expected pretty msg to be ${expectedPrettyMsg}, found ${actualPrettyMsg}`
+                        )
+                        assert.deepStrictEqual(
+                            expectedErrorTokens,
+                            actualErrorTokens,
+                            `expected error tokens to be ${expectedErrorTokens}, found ${actualErrorTokens}`
+                        )
+                    } catch (error) {
+                        assert.fail(`Error testing ${level} level: ${String(error)}`)
+                    }
+                })
+            })
+        })
+    }
+
+    const testData = [
+        {
+            title: 'logs w/o template params',
+            nlsKey: 'silly.key1',
+            nlsTemplate: 'Yay',
+            templateTokens: undefined,
+        },
+        {
+            title: 'logs with 1 string template param',
+            nlsKey: 'silly.key2',
+            nlsTemplate: 'Nice to meet you "{0}"',
+            templateTokens: ['bob'],
+        },
+        {
+            title: 'logs with 2 string template params',
+            nlsKey: 'silly.key3',
+            nlsTemplate: 'Hey "{0}", meet "{1}"',
+            templateTokens: ['bob', 'joe'],
+        },
+        {
+            title: 'logs with 3 string template params',
+            nlsKey: 'silly.key4',
+            nlsTemplate: 'Hey "{0}", meet "{1}" and "{2}"',
+            templateTokens: ['bob', 'joe', 'kim'],
+        },
+        {
+            title: 'logs with 2 template params: errro, string',
+            nlsKey: 'silly.key5',
+            nlsTemplate: 'Oh no "{1}", we found an error: "{0}"',
+            templateTokens: [new Error('Stock market crash'), 'joe'],
+        },
+        {
+            title: 'logs with 2 template params: error, error',
+            nlsKey: 'silly.key6',
+            nlsTemplate: '1st Error "{0}"; 2nd error: "{1}"',
+            templateTokens: [new Error('Error zero'), new Error('Error one')],
+        },
+        {
+            title: 'logs with 3 template params: string, error, error',
+            nlsKey: 'silly.key7',
+            nlsTemplate: 'Oh my "{0}", there are errors: 1st Error "{0}"; 2nd error: "{1}"',
```

Comment:
Shouldn't 1st error use {1} and 2nd error {2} ?

Reply:
✔️